pandas/core/dtypes/cast.py

```diff
@@ -1094,14 +1093,12 @@ def astype_nansafe(
         The dtype was a datetime64/timedelta64 dtype, but it had no unit.
     """
     if arr.ndim > 1:
-        # Make sure we are doing non-copy ravel and reshape.
-        flags = arr.flags
-        flat = arr.ravel("K")
+        # TODO: try to use contiguity to avoid potentially copying here, see #42475
```

Comment:
maybe lose this comment. trying to do this caused the regression.

Reply:
have removed